src/Features/CSharp/Portable/CodeFixes/Async/CSharpAddAwaitCodeFixProvider.cs

```diff
@@ -76,8 +72,19 @@ private bool IsCorrectReturnType(ExpressionSyntax expression, SemanticModel sema
 
         private static ExpressionSyntax ConvertToAwaitExpression(ExpressionSyntax expression)
         {
-            return SyntaxFactory.AwaitExpression(expression)
-                                .WithAdditionalAnnotations(Formatter.Annotation);
+            AwaitExpressionSyntax result;
+
+            if (expression.HasLeadingTrivia)
```

Comment:
>"Shouldn't the first line of your code be ..(expression.WithoutTrivia())?"

`WithTriviaFrom(expression)` will override the trivia on that node.

 >For example:
```
- return true ? Task.Delay(0) /* 0 */ : Task.Delay(1) /* 1 */;
+ return await (true ? Task.Delay(0) /* 0 */ : Task.Delay(1)) /* 1 */;
```

I don't understand this case.  This is not going to be valid code.  Are you suggesting we offer a fix here?  What does the containing method for this look like?  

>The comment looks wrongly propagated for me

Handling trivia is difficult.  At the very least we should attempt to keep all the users trivia for them, giving them the ability to fix any sections that get misplaced.